lib/src/visitor/async_evaluate.dart

```diff
@@ -2472,6 +2476,9 @@ final class _EvaluateVisitor
       }
       throwWithTrace(_exception(error.message, node.span), error, stackTrace);
     }
+    _callableNode = oldCallableNode;
```

Comment:
If you put this in a `finally` clause on the `try`, you can avoid needing to create the intermediate `result` variable.

Reply:
I thought if we're going to throw there is no need to set the variable back to original value, but styling wise finally block looks better.

Reply:
`finally` gets run even if nothing throws, so it's useful for cases like these where you want to do something _after_ a return.